api/tests/unit/domain/events/compute-validated-skills-count_test.js

```diff
@@ -0,0 +1,83 @@
+const computeValidatedSkillsCount = require('../../../../lib/domain/events/compute-validated-skills-count');
+const CampaignParticipationResultsShared = require('./../../../../lib/domain/events/CampaignParticipationResultsShared');
+
+describe('Unit | Domain | Events | compute-validated-skills-count', function() {
+
+  describe('eventType', () => {
+    it('returns the CampaignParticipationResultsShared', () => {
+      const eventType = computeValidatedSkillsCount.eventType;
+
+      expect(eventType).to.equals(CampaignParticipationResultsShared);
+    });
+  });
+
+  describe('.computeValidatedSkillsCount', () => {
+    let knowledgeElementRepository;
+    let campaignParticipationRepository;
+    let targetProfileWithLearningContentRepository;
+
+    beforeEach(() => {
+      campaignParticipationRepository = { get: sinon.stub(), update: sinon.stub() };
+      targetProfileWithLearningContentRepository = { getByCampaignId: sinon.stub() };
+      knowledgeElementRepository = { countValidatedTargetedByCompetencesForOneUser: sinon.stub() };
+    });
+
+    context('when the campaign is of type assessment', () => {
+      it('computes the CampaignParticipationResultsShared', async () => {
+        const targetProfile = domainBuilder.buildTargetProfile();
+        const campaign = domainBuilder.buildCampaign.ofTypeAssessment();
+        const campaignParticipation = domainBuilder.buildCampaignParticipation({ id: 1, campaignId: 2, userId: 3, campaign });
+        const event = new CampaignParticipationResultsShared({ campaignParticipationId: campaignParticipation.id });
+        const validatedSkillsCountByCompetence = {
+          'cpmpetence1': 2,
+          'cpmpetence2': 3,
+        };
+        const validatedSkillsCount = 5;
+
+        sinon.stub(campaignParticipation, 'setResults');
+
+        campaignParticipationRepository.get
+          .withArgs(event.campaignParticipationId, { include: ['campaign'] })
+          .resolves(campaignParticipation);
+
+        targetProfileWithLearningContentRepository.getByCampaignId
+          .withArgs({ campaignId: campaignParticipation.campaignId })
+          .resolves(targetProfile);
+
+        knowledgeElementRepository.countValidatedTargetedByCompetencesForOneUser
+          .withArgs(campaignParticipation.userId, campaignParticipation.sharedAt, targetProfile)
+          .resolves(validatedSkillsCountByCompetence);
+
+        await computeValidatedSkillsCount({ event, campaignParticipationRepository, knowledgeElementRepository, targetProfileWithLearningContentRepository });
+
+        expect(campaignParticipation.setResults).to.have.been.calledWith(validatedSkillsCount);
+        expect(campaignParticipationRepository.update).to.have.been.calledWith(campaignParticipation);
+      });
+    });
+
+    context('when the campaign is of type profile collection', () => {
+      it('computes the CampaignParticipationResultsShared', async () => {
```

Comment:
`does not compute the CampaignParticipationResultsShared`

Reply:
validatedSkillsCount

Reply:
done